src/App.js

```diff
@@ -7,10 +7,13 @@ import AllProducts from './components/AllProducts';
 import Cart from './components/Cart';
 import categories from './categories.json';
 import { useEffect, useState } from 'react';
+import AppContext from './components/context';
```

Comment:
make it separate changes:

- adding AppContext
- Fixing favorites

Also add detailed description what is changed and why it's needed